pkg/controller/controller.go

```diff
@@ -220,12 +238,24 @@ func New(ctx context.Context, cfg *config.Config, kubeclientset kubernetes.Inter
 		wfLauncher = launchplan.NewFailFastLaunchPlanExecutor()
 	}
 
+	namespaceFilterR := func() *regexp.Regexp {
+		if cfg.NamespaceFilter != "" {
+			r, err := regexp.Compile(cfg.NamespaceFilter)
```

Comment:
If you opt for the config type (first comment), you are guaranteed a valid regex at this point...